### PR TITLE
LegalDocuments: Fix filter, pagination, self registration, change messages

### PR DESCRIPTION
**Pull request by lscharmer**
This PR fixes the following mantis issues:
- \#39385
- \#39387
- \#39414
- \#39415
- \#39418
- \#39422
- \#39477
- \#39667
- \#39668
- \#39670
- \#39690
- \#39691
- \#39692
- \#39693
- \#39696
- \#39706

Additionally the following things are changed:
- Deletion of a document doesn't hide all the corresponding versions in the "Acceptance History" Tab.
- Updating a document will not trigger a resigning of the document. This is now the **same** behaviour as in ILIAS8.
- Deleting all documents will automatically disable the service (ToS or Dpro). This is now the **same** behaviour as in ILIAS 8.

### CODE REVIEW

File: lang/ilias_en.lang [hunk 1]
```diff
@@ -8827,7 +8827,7 @@ didactic#:#sess_closed_info#:#Session is not visible for non-participants.
 dpro#:#dpro_account_reg_not_possible#:#Self-registration is currently not possible because there is no Data Protection agreement available. Please contact your <a href="%1$s">system administrator</a> for further information.
 dpro#:#dpro_agree_date#:#Data Protection agreed on
 dpro#:#dpro_last_reset_date#:#The Data Protection were reset on %s. Only reset the Data Protection if changes have been made to the document(s) and you require all users to agree to the documents.
-dpro#:#dpro_mode#:#Mode
+dpro#:#dpro_mode#:#Agreement mode
```
Comment (mjansenDatabay) [inline lang/ilias_en.lang:8830]:
```suggestion
dpro#:#dpro_mode#:#Agreement Mode
```

File: Services/LegalDocuments/classes/ConsumerToolbox/ConsumerSlots/Agreement.php [hunk 2]
```diff
@@ -61,17 +61,21 @@ public function showAgreement(string $gui, string $cmd): PageFragment
 
     public function showAgreementForm(string $gui, string $cmd): PageFragment
     {
+        $form = $this->user->matchingDocument()->isOk() ?
+              $this->agreementForm($gui, $cmd) :
+              $this->ui->create()->legacy('<br/><br/>');
```
Comment (mjansenDatabay) [inline Services/LegalDocuments/classes/ConsumerToolbox/ConsumerSlots/Agreement.php]: Can we replace this by a divider or something else from the UI framework?

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]

File: Services/LegalDocuments/classes/ConsumerToolbox/ConsumerSlots/Agreement.php (hunk 2 above)
Comment (lscharmer) [reply Services/LegalDocuments/classes/ConsumerToolbox/ConsumerSlots/Agreement.php]: Ok.

File: lang/ilias_en.lang (hunk 1 above)
Comment (lscharmer) [reply lang/ilias_en.lang:8830]: Is changed.

File: Services/LegalDocuments/classes/ConsumerToolbox/ConsumerSlots/Agreement.php (hunk 2 above)
Comment (lscharmer) [reply Services/LegalDocuments/classes/ConsumerToolbox/ConsumerSlots/Agreement.php]: Is changed to a divider.

**Review by mjansenDatabay — APPROVED** [review]

**Pull request merged**